configure.ac

```diff
@@ -26,6 +26,8 @@ AM_CONDITIONAL([WIN32], test "$WIN32" = "yes")
 AC_CHECK_FUNCS([wprintf])
 AC_CHECK_FUNCS([reallocarray])  # needs AC_USE_SYSTEM_EXTENSIONS (#define _GNU_SOURCE)
 
+## Haiku check for -lsocket -lnetwork inet_ntop
+AC_SEARCH_LIBS([socket],[socket inet network])
```

Comment:
The [signature of `AC_SEARCH_LIBS`](https://www.gnu.org/software/autoconf/manual/autoconf-2.66/html_node/Libraries.html) is …
> `AC_SEARCH_LIBS (function, search-libs, [action-if-found], [action-if-not-found], [other-libraries])`

… so the use above would pass `socket` for parameter `function`.
Could it be we want `inet_ntop` for `function`, instead?

Reply:
Also, is `inet` needed? The comment line implies otherwise, in a way.

Reply:
Need to review this, I'm not a developer so I used a base from another recipe, if you have a better suited solution I'm happy to hear about it :) I'll get back on this when I have the time again to look more into it

Reply:
Not a better solution, no. I would need a running Haiku to debug this myself, just noticed that there is a difference between comment and implementation. Please keep me updated, thank you!

Reply:
It "looks better" but does it fix the problem? I cannot tell.

Reply:
Just checked, on gcc2hybrid (my main Haiku install):
For gcc2 it's using -lsocket
For gcc7 it's using -lnetwork to link against
Both are working ok to build and create the package and passes the tests (although I don't see any actual test being performed with "make check"?)

Reply:
> For gcc2 it's using -lsocket
> For gcc7 it's using -lnetwork to link against

Sounds excellent, thank you!

> (although I don't see any actual test being performed with "make check"?)

`make check` should be running tests. The code depends on C++98 and GTest >=1.8.1. Maybe some part of the Haiku packaging is cutting that code away for you?

Reply:
PS: I'll need to convert this to CMake for #47 soon it seems. Where can I find the current version of CMake in Haiku?

Reply:
PS: But let's first merge the Autotools version...

Reply:
PS: My current CMake working draft at #47 is this:
```cmake
if(HAIKU)
    # Function inet_ntop needs -lsocket or -lnetwork (see pull request #45)
    check_library_exists(socket inet_ntop "" HAVE_LIBSOCKET__INET_NTOP)
    check_library_exists(network inet_ntop "" HAVE_LIBNETWORK__INET_NTOP)
    if(HAVE_LIBSOCKET__INET_NTOP)
        target_link_libraries(uriparse PUBLIC socket)
    endif(HAVE_LIBSOCKET__INET_NTOP)
    if(HAVE_LIBNETWORK__INET_NTOP)
        target_link_libraries(uriparse PUBLIC network)
    endif(HAVE_LIBNETWORK__INET_NTOP)
endif(HAIKU)
```

Reply:
I'll  look into it today :)